TestWebsocketLargeCall: timeout on CI

`ubuntu` and `mac` CI
```
panic: test timed out after 1h0m0s
	running tests:
		TestWebsocketLargeCall (59m53s)

goroutine 10613 [running]:
testing.(*M).startAlarm.func1()
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/testing/testing.go:2484 +0x308
created by time.goFunc
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/time/sleep.go:215 +0x38

goroutine 1 [chan receive, 59 minutes]:
testing.tRunner.func1()
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/testing/testing.go:1753 +0x43c
testing.tRunner(0x140002541c0, 0x140000edc68)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/testing/testing.go:1798 +0x120
testing.runTests(0x1400022a300, {0x103851380, 0x28, 0x28}, {0x1400020a680?, 0x7?, 0x1038a4620?})
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/testing/testing.go:2277 +0x3ec
testing.(*M).Run(0x14000250140)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/testing/testing.go:2142 +0x588
main.main()
	_testmain.go:139 +0x90

goroutine 9681 [IO wait, 59 minutes]:
internal/poll.runtime_pollWait(0x14a6da708, 0x72)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/runtime/netpoll.go:351 +0xa0
internal/poll.(*pollDesc).wait(0x140003c8280?, 0x102bf92a8?, 0x0)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/internal/poll/fd_poll_runtime.go:84 +0x28
internal/poll.(*pollDesc).waitRead(...)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0x140003c8280)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/internal/poll/fd_unix.go:620 +0x24c
net.(*netFD).accept(0x140003c8280)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x1400092e180)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/tcpsock_posix.go:159 +0x24
net.(*TCPListener).Accept(0x1400092e180)
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/tcpsock.go:380 +0x2c
net/http.(*Server).Serve(0x1400024a100, {0x1033f7730, 0x1400092e180})
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/http/server.go:3424 +0x290
net/http/httptest.(*Server).goServe.func1()
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/http/httptest/server.go:311 +0x5c
created by net/http/httptest.(*Server).goServe in goroutine 9697
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/net/http/httptest/server.go:309 +0x6c

goroutine 10407 [chan receive, 59 minutes]:
github.com/erigontech/erigon/rpc.(*testService).Block(0x140001b5508?, {0x1033f8880?, 0x140006bc870?})
	/Users/runner/work/erigon/erigon/rpc/testservice_test.go:97 +0x34
reflect.Value.call({0x1400023e370?, 0x1400006e3b0?, 0x1400011db38?}, {0x1031a18de, 0x4}, {0x140002f6450, 0x2, 0x103135408?})
	/Users/runner/work/erigon/erigon/rpc/handler.go:240 +0x1c8
github.com/erigontech/erigon/rpc.(*handler).startCallProc.func1()
	/Users/runner/work/erigon/erigon/rpc/handler.go:333 +0xc4
created by github.com/erigontech/erigon/rpc.(*handler).startCallProc in goroutine 10215
	/Users/runner/work/erigon/erigon/rpc/handler.go:329 +0xb0

goroutine 10244 [chan receive, 59 minutes]:
github.com/erigontech/erigon/rpc.(*testService).Block(0x1400015f508?, {0x1033f8880?, 0x140002da690?})
	/Users/runner/work/erigon/erigon/rpc/testservice_test.go:97 +0x34
reflect.Value.call({0x1400023e370?, 0x1400006e3b0?, 0x1400007ab38?}, {0x1031a18de, 0x4}, {0x140002f7cb0, 0x2, 0x103135408?})
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/reflect/value.go:584 +0x978
reflect.Value.Call({0x1400023e370?, 0x1400006e3b0?, 0x1400007abe8?}, {0x140002f7cb0?, 0x1400007abb8?, 0x102d293b0?})
	/Users/runner/hostedtoolcache/go/1.24.6/arm64/src/reflect/value.go:368 +0x94
github.com/erigontech/erigon/rpc.(*callback).call(0x140003445b0, {0x1033f8880, 0x140002da690}, {0x140004b4300, 0xa}, {0x0, 0x0, 0x68afab62?}, {0x103402a88, 0x1400090a4c0})
	/Users/runner/work/erigon/erigon/rpc/service.go:227 +0x42c
github.com/erigontech/erigon/rpc.(*handler).runMethod(0x0?, {0x1033f8880?, 0x140002da690?}, 0x140003444d0, 0x0?, {0x0?, 0x1038cd6c0?, 0x1400007adc8?}, {0x103402a88?, 0x1400090a4c0?})
	/Users/runner/work/erigon/erigon/rpc/handler.go:529 +0x26c
github.com/erigontech/erigon/rpc.(*handler).handleCall(0x14000427110, 0x140002f7c50, 0x140003444d0, {0x103402a88, 0x1400090a4c0})
	/Users/runner/work/erigon/erigon/rpc/handler.go:479 +0x3c8
github.com/erigontech/erigon/rpc.(*handler).handleCallMsg(0x14000427110, 0x140002f7c50, 0x140003444d0, {0x103402a88, 0x1400090a4c0})
	/Users/runner/work/erigon/erigon/rpc/handler.go:420 +0x474
github.com/erigontech/erigon/rpc.(*handler).handleMsg.func1(0x140002f7c50)
	/Users/runner/work/erigon/erigon/rpc/handler.go:240 +0x1c8
github.com/erigontech/erigon/rpc.(*handler).startCallProc.func1()
	/Users/runner/work/erigon/erigon/rpc/handler.go:333 +0xc4
created by github.com/erigontech/erigon/rpc.(*handler).startCallProc in goroutine 10215
	/Users/runner/work/erigon/erigon/rpc/handler.go:329 +0xb0

goroutine 10208 [select, 59 minutes]:
github.com/erigontech/erigon/rpc.(*Client).dispatch(0x140001a81e0, {0x1033fae78, 0x14000416000})
	/Users/runner/work/erigon/erigon/rpc/client.go:563 +0x1fc
created by github.com/erigontech/erigon/rpc.initClient in goroutine 73
	/Users/runner/work/erigon/erigon/rpc/client.go:231 +0x338
```
https://github.com/erigontech/erigon/actions/runs/17282535173/job/49053619083#step:7:553

--- a/rpc/client.go
+++ b/rpc/client.go
@@ -592,6 +592,9 @@ func (c *Client) dispatch(codec ServerCodec) {
 			conn.handler.logger.Trace("RPC connection read error", "err", err)
 			// A read error is fatal for the connection, and all pending requests must be cancelled, including any
 			// that might still be considered in-flight.
+			if lastOp != nil {
+				conn.handler.removeRequestOp(lastOp)
+			}
 			conn.close(err, nil)
 			reading = false
 

--- a/rpc/websocket_test.go
+++ b/rpc/websocket_test.go
@@ -85,10 +85,6 @@ func TestWebsocketOriginCheck(t *testing.T) {
 
 // This test checks whether calls exceeding the request size limit are rejected.
 func TestWebsocketLargeCall(t *testing.T) {
-	//if runtime.GOOS == "darwin" {
-	t.Skip("issue #16875")
-	//}
-
 	if testing.Short() {
 		t.Skip()
 	}

